src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutorExtensions.cs

```diff
@@ -36,7 +36,6 @@ public static async Task<IDelayedException> TryExecuteAsync(this IFunctionExecut
                 // Set retry context only if retry strategy exists and this is not first attempt
                 if (functionInstance.FunctionDescriptor.RetryStrategy != null && attempt > 0 && functionInstance is FunctionInstance instance)
                 {
-                    retryContext = retryContext ?? new RetryContext();
```

Comment:
I see we can also simplify the above if clause - we only need `if(retryContext != null && functionInstance is FunctionInstance instance)` since attempt will always be > 0 at that point, and we short circuit below if RetryStrategy is null.